File: app/src/main/java/com/app/grocerydemo/service/APIService.java
package com.app.grocerydemo.service;


import com.app.grocerydemo.model.LoginMainModel;
import com.app.grocerydemo.model.MainListDataModel;

import java.util.Map;

import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;
import retrofit2.http.Path;
import retrofit2.http.Query;
import retrofit2.http.Url;


public interface APIService {

    @POST("api/login")
    Call<LoginMainModel> loginUser(@Body Map<String, String> body);


    @GET("api/LocationClockInTypeSetting")
    Call<LoginMainModel> LocationClockInTypeSetting(@Header("Authorization") String token,
                                                    @Query("ClientCode") String ClientCode,
                                                    @Query("LocationId") String LocationId);

    @POST("api/EmployeeClockInVerification")
    Call<LoginMainModel> EmployeeClockInVerification(@Header("Authorization") String token,
                                                     @Body Map<String, String> body);

    @Multipart
    @POST("api/EmployeeTimesheet")
    Call<LoginMainModel> EmployeeTimesheet(@Header("Authorization") String accesstoken,
                                           @Part("EmployeeId") RequestBody employeeId,
                                           @Part("LocationId") RequestBody locationId,
                                           @Part("ClockInTypeId") RequestBody clockInTypeId,
                                           @Part("ClockInOutType") RequestBody clockInOutType,
                                           @Part("ClockInTime") RequestBody clockInTime,
                                           @Part("ClientCode") RequestBody clientCode,
                                           @Part("DeviceType") RequestBody deviceType,
                                           @Part MultipartBody.Part bodyImage);

    @Multipart
    @POST("api/EmployeeTimesheet/SaveEmployeeFaceImage")
    Call<ResponseBody> saveEmployeeFaceImage(@Header("Authorization") String accesstoken,
                                               @Part("ClientCode") RequestBody clientCode,
                                               @Part("LocationId") RequestBody locationId,
                                               @Part("EmployeeId") RequestBody employeeId,
                                               @Part("PersonGroupId") RequestBody PersonGroupId,
                                               @Part MultipartBody.Part bodyImage);

    @POST("api/EmployeeClockInVerification/VerifyQRCode")
    Call<LoginMainModel> VerifyQRCode(@Header("Authorization") String token,
                                      @Body Map<String, String> body);

    //======================== Rota app ===================================//


//    @GET("find/{key}")
//    Call<FindAddressModel> findAddress(@Header("api-key") String key,
//                                       @Path("key") String key_val);


    @POST("api/Login")
    Call<LoginMainModel> login(@Body Map<String, String> body);


    @GET("api/EmployeeRota/GetClockInData")
    Call<MainListDataModel> getClockInData(@Header("Authorization") String token,
                                           @Query("EmployeeId") String EmployeeId,
                                           @Query("ClientCode") String ClientCode,
                                           @Query("Startdate") String Startdate,
                                           @Query("Enddate") String Enddate);

    @Multipart
    @POST("api/EmployeeProfilePage")
    Call<LoginMainModel> uploadEmployeeProfilePage(@Header("Authorization") String token,
                                                   @Part("ClientCode") RequestBody ClientCode,
                                                   @Part("EmployeeId") RequestBody EmployeeId,
                                                   @Part MultipartBody.Part files);


    @Multipart
    @POST("api/EmployeeDocument")
    Call<LoginMainModel> uploadEmpDoc(@Header("Authorization") String token,
                                      @Part("ClientCode") RequestBody ClientCode,
                                      @Part("EmployeeId") RequestBody EmployeeId,
                                      @Part("TypeId") RequestBody TypeId,
                                      @Part("LocationId") RequestBody LocationId,
                                      @Part("FileDescription") RequestBody FileDescription,
                                      @Part MultipartBody.Part files);

    @FormUrlEncoded
    @POST("api/EmployeeDocument")
    Call<LoginMainModel> editEmpDoc(@Header("Authorization") String token,
                                    @Field("ClientCode") String ClientCode,
                                    @Field("EmployeeId") String EmployeeId,
                                    @Field("TypeId") String TypeId,
                                    @Field("FileDescription") String FileDescription,
                                    @Field("UploadFileId") String uploadFileId);

    @DELETE("api/EmployeeDocument")
    Call<LoginMainModel> deleteEmpDoc(@Header("Authorization") String token,
                                      @Query("ClientCode") String ClientCode,
                                      @Query("uploadfileId") String uploadfileId,
                                      @Query("filePath") String filePath);

    @GET("api/EmployeeDocument")
    Call<LoginMainModel> getEmployeeDocumentEdit(@Header("Authorization") String token,
                                                 @Query("EmployeeId") String EmployeeId,
                                                 @Query("ClientCode") String ClientCode,
                                                 @Query("UploadFileId") String UploadFileId);


    @GET
    Call<ResponseBody> downloadUrl(@Url String url);

    @GET("api/EmployeeDocument/DownloadFile")
    Call<ResponseBody> DownloadFile(@Query("filePath") String url);

}

File: app/src/main/java/com/app/grocerydemo/kotlin_demo/service/APIService.kt
package com.app.grocerydemo.kotlin_demo.service

import com.app.grocerydemo.model.LoginMainModel
import okhttp3.RequestBody
import okhttp3.MultipartBody
import okhttp3.ResponseBody
import com.app.grocerydemo.model.MainListDataModel
import retrofit2.Call
import retrofit2.http.*

interface APIService {
    @POST("api/login")
    fun loginUser(@Body body: Map<String?, String?>?): Call<LoginMainModel?>?

    @GET("api/LocationClockInTypeSetting")
    fun LocationClockInTypeSetting(
        @Header("Authorization") token: String?,
        @Query("ClientCode") ClientCode: String?,
        @Query("LocationId") LocationId: String?
    ): Call<LoginMainModel?>?

    @POST("api/EmployeeClockInVerification")
    fun EmployeeClockInVerification(
        @Header("Authorization") token: String?,
        @Body body: Map<String?, String?>?
    ): Call<LoginMainModel?>?

    @Multipart
    @POST("api/EmployeeTimesheet")
    fun EmployeeTimesheet(
        @Header("Authorization") accesstoken: String?,
        @Part("EmployeeId") employeeId: RequestBody?,
        @Part("LocationId") locationId: RequestBody?,
        @Part("ClockInTypeId") clockInTypeId: RequestBody?,
        @Part("ClockInOutType") clockInOutType: RequestBody?,
        @Part("ClockInTime") clockInTime: RequestBody?,
        @Part("ClientCode") clientCode: RequestBody?,
        @Part("DeviceType") deviceType: RequestBody?,
        @Part bodyImage: MultipartBody.Part?
    ): Call<LoginMainModel?>?

    @Multipart
    @POST("api/EmployeeTimesheet/SaveEmployeeFaceImage")
    fun saveEmployeeFaceImage(
        @Header("Authorization") accesstoken: String?,
        @Part("ClientCode") clientCode: RequestBody?,
        @Part("LocationId") locationId: RequestBody?,
        @Part("EmployeeId") employeeId: RequestBody?,
        @Part("PersonGroupId") PersonGroupId: RequestBody?,
        @Part bodyImage: MultipartBody.Part?
    ): Call<ResponseBody?>?

    @POST("api/EmployeeClockInVerification/VerifyQRCode")
    fun VerifyQRCode(
        @Header("Authorization") token: String?,
        @Body body: Map<String?, String?>?
    ): Call<LoginMainModel?>?

    //======================== Rota app ===================================//
    //    @GET("find/{key}")
    //    Call<FindAddressModel> findAddress(@Header("api-key") String key,
    //                                       @Path("key") String key_val);
    @POST("api/Login")
    fun login(@Body body: Map<String?, String?>?): Call<LoginMainModel?>?

    @GET("api/EmployeeRota/GetClockInData")
    fun getClockInData(
        @Header("Authorization") token: String?,
        @Query("EmployeeId") EmployeeId: String?,
        @Query("ClientCode") ClientCode: String?,
        @Query("Startdate") Startdate: String?,
        @Query("Enddate") Enddate: String?
    ): Call<MainListDataModel?>?

    @Multipart
    @POST("api/EmployeeProfilePage")
    fun uploadEmployeeProfilePage(
        @Header("Authorization") token: String?,
        @Part("ClientCode") ClientCode: RequestBody?,
        @Part("EmployeeId") EmployeeId: RequestBody?,
        @Part files: MultipartBody.Part?
    ): Call<LoginMainModel?>?

    @Multipart
    @POST("api/EmployeeDocument")
    fun uploadEmpDoc(
        @Header("Authorization") token: String?,
        @Part("ClientCode") ClientCode: RequestBody?,
        @Part("EmployeeId") EmployeeId: RequestBody?,
        @Part("TypeId") TypeId: RequestBody?,
        @Part("LocationId") LocationId: RequestBody?,
        @Part("FileDescription") FileDescription: RequestBody?,
        @Part files: MultipartBody.Part?
    ): Call<LoginMainModel?>?

    @FormUrlEncoded
    @POST("api/EmployeeDocument")
    fun editEmpDoc(
        @Header("Authorization") token: String?,
        @Field("ClientCode") ClientCode: String?,
        @Field("EmployeeId") EmployeeId: String?,
        @Field("TypeId") TypeId: String?,
        @Field("FileDescription") FileDescription: String?,
        @Field("UploadFileId") uploadFileId: String?
    ): Call<LoginMainModel?>?

    @DELETE("api/EmployeeDocument")
    fun deleteEmpDoc(
        @Header("Authorization") token: String?,
        @Query("ClientCode") ClientCode: String?,
        @Query("uploadfileId") uploadfileId: String?,
        @Query("filePath") filePath: String?
    ): Call<LoginMainModel?>?

    @GET("api/EmployeeDocument")
    fun getEmployeeDocumentEdit(
        @Header("Authorization") token: String?,
        @Query("EmployeeId") EmployeeId: String?,
        @Query("ClientCode") ClientCode: String?,
        @Query("UploadFileId") UploadFileId: String?
    ): Call<LoginMainModel?>?

    @GET
    fun downloadUrl(@Url url: String?): Call<ResponseBody?>?

    @GET("api/EmployeeDocument/DownloadFile")
    fun DownloadFile(@Query("filePath") url: String?): Call<ResponseBody?>?
}